modules/mauth-authenticator/src/test/scala/com/mdsol/mauth/scaladsl/utils/RequestAuthenticatorSpec.scala

```diff
@@ -72,6 +72,47 @@ class RequestAuthenticatorSpec extends FlatSpec with RequestAuthenticatorBaseSpe
     }
   }
 
+  it should "authenticate a valid request with V2 headers" in clientContext { (client) =>
+    //noinspection ConvertibleToMethodValue
+    (mockEpochTimeProvider.inSeconds _: () => Long).expects().returns(CLIENT_X_MWS_TIME_HEADER_VALUE.toLong + 3)
+    val authenticator = new RequestAuthenticator(client, mockEpochTimeProvider)
+
+    whenReady(authenticator.authenticate(getSimpleRequestV2)) { validationResult =>
+      validationResult shouldBe true
+    }
+  }
+
+  it should "authenticate a valid request with V2 headers if disabled V1" in clientContext { (client) =>
+    //noinspection ConvertibleToMethodValue
+    (mockEpochTimeProvider.inSeconds _: () => Long).expects().returns(CLIENT_X_MWS_TIME_HEADER_VALUE.toLong + 3)
+    val authenticator = new RequestAuthenticator(client, mockEpochTimeProvider, true)
+
+    whenReady(authenticator.authenticate(getSimpleRequestV2)) { validationResult =>
+      validationResult shouldBe true
+    }
+  }
+
+  it should "authenticate a valid request with all headers" in clientContext { (client) =>
+    //noinspection ConvertibleToMethodValue
```

Comment:
what is this for?

Reply:
This is for verify with the both v1 and v2 headers provided. I have renamed as "authenticate a valid request with the both V1 and V2 headers provided"

Reply:
I meant the comment :)